jwcrypto/jwe.py

```diff
@@ -635,6 +701,20 @@ def serialize(self, compact=False):
                 rec = self.objects['recipients'][0]
             else:
                 rec = self.objects
+            if 'header' in rec:
```

Comment:
Seem the least invasive one, all other alternatives I could think of were more invasive and as ugly.